### ISSUE TITLE
[FEATURE] Jetbrains Rider Support

### ISSUE THREAD

**Issue by harrypooni-118** [DataDog/datadog-for-intellij-platform#34]
## Describe the solution you'd like

Plugin to available for Jetbrains Rider
## Describe alternatives you've considered

N/A
## Additional context

Other Jetbrains products supported.

**Comment by Vomain** [comment]
@harrypooni-118 out of curiosity, what feature of the DD plugin are you particularly interested in?

**Comment by harrypooni-118** [comment]
@Vomain APM and Logs are the main features that i am looking for. From a development perspective, analysing trace information to resolve an issue is key for faster delivery.

**Comment by matejukx** [comment]
+1
Would love to use SAST features directly in Rider

**Comment by agarbutt** [comment]
My teams would really value the Profiling, Logging, **ASM**, and **Static Analysis tooling**.

**Comment by tystol** [comment]
+approx20
The vast majority of devs in my teams now use Rider over VS, and we would love to see our product's APM/profiling/log data directly in the IDE

**Comment by activist** [comment]
https://plugins.jetbrains.com/plugin/19495-datadog/reviews
> Compatible with IntelliJ IDEA, GoLand, PhpStorm, PyCharm, RubyMine, WebStorm

Isn't it possible to enable the plugin altogether for Rider as well?

**Comment by abrooksv** [comment]
> https://plugins.jetbrains.com/plugin/19495-datadog/reviews
> 
> > Compatible with IntelliJ IDEA, GoLand, PhpStorm, PyCharm, RubyMine, WebStorm
> 
> Isn't it possible to enable the plugin altogether for Rider as well?

Yes, but many of the features aren't supported in Rider. Some stuff will work, but anything that interacts with the source code will not. 

For example, you could use Error Tracking tab to list errors for your service identified by the Error Tracking product, but you won't be able to navigate to the source code from the stack traces.
Other things like Log inlays and stuff will not work since the language integrations are not in place.

For a bit of background on why Rider is complicated: It is not architected the same as any other IntelliJ-platform based IDE. There is the standard Java-based IDE frontend, but all the language parsing and interactions happens out of process via a ReSharper plugin and communicated back to the Java UI via an RPC system.
